src/autocomplete/QueryMatcher.ts

```diff
@@ -49,7 +49,7 @@ export default class QueryMatcher<T extends Object> {
     private _options: IOptions<T>;
     private _keys: IOptions<T>["keys"];
     private _funcs: Required<IOptions<T>["funcs"]>;
-    private _items: Map<string, T[]>;
+    private _items: Map<{value: string, weight: number}, T[]>;
```

Comment:
instead of bloating this map by splitting the keys into {key,index}
could we not optimize given that we only actually ever care about the "highest priority" (lowest weight) and thus just store the Min(weight) in the Value (along with `T[]`) that we find when building the _items map?